Libraries/LibGUI/MultiView.cpp

```diff
@@ -106,9 +106,9 @@ MultiView::MultiView()
             on_drop(index, event);
     };
 #endif
-    set_view_mode(ViewMode::Icon);
 
     build_actions();
+    m_view_as_icons_action->set_checked(true);
```

Comment:
Why was `set_view_mode(ViewMode::Icon)` not making the button checked?
I think this change worsens code readability, as we are replacing a high-level operation ("go into icon view mode") with a low-level operation ("change the checked state of this specific button") so it's less obvious that everything will be set up correctly afterwards.

Reply:
Good point.

`m_view_as_icons_action->set_checked` calls `set_view_mode`. Meanwhile, `set_view_mode` is a public method, and can be called from the outside.
So I guess the best way around this is calling `action->set_checked()` from `set_view_mode`, and stopping the recursion by this already existing snippet in `set_view_mode`:
```
    if (m_view_mode == mode)
        return;
    m_view_mode = mode;
```

What do you think about that?

Reply:
Fun fact: FileManager also initializes its state by calling [`action->set_checked(true);`](https://github.com/SerenityOS/serenity/blob/46b79eaad9389e438d99bc06e5beca4408abb089/Applications/FileManager/main.cpp#L837)